### PR TITLE
support load variables from env & debugging mode

### PR DESCRIPTION
**Pull request by jianyexi**

### CODE REVIEW

File: lib/testScenario/postmanCollectionRunnerClient.ts [hunk 1]
```diff
@@ -508,7 +535,18 @@ export class PostmanCollectionRunnerClient implements TestScenarioRunnerClient {
...
+          runtimeEnvManager.save(_summary.item.name, this, "beforeStep");
+        }
+      })
+      .on("item", async function (this: any, _err, _summary) {
+        if (!_err) {
+          runtimeEnvManager.clean();
+        }
+      })
       .on("done", async (_err, _summary) => {
+        //sinkRuntimeEnv();
```
Comment (ruowan) [inline lib/testScenario/postmanCollectionRunnerClient.ts]: what's sinkRuntimeEnv

Comment (jianyexi) [reply lib/testScenario/postmanCollectionRunnerClient.ts]: I 'll remove it

File: lib/testScenario/postmanCollectionRunnerClient.ts [hunk 2]
```diff
@@ -508,6 +534,16 @@ export class PostmanCollectionRunnerClient implements TestScenarioRunnerClient {
...
         }
       )
+      .on("beforeItem", async function (this: any, _err, _summary) {
+        if (!_err) {
+          runtimeEnvManager.save(_summary.item.name, this, "beforeStep");
+        }
+      })
+      .on("item", async function (this: any, _err, _summary) {
+        if (!_err) {
+          runtimeEnvManager.clean();
```
Comment (ruowan) [inline lib/testScenario/postmanCollectionRunnerClient.ts:544]: Why do clean here

File: lib/testScenario/runtimeEnvManager.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    const fromStep = this.opts.from || this.opts.to ? this.collection.items.idx(0).name : undefined;
+    if (fromStep) {
+      this.cleanEnv(this.collection, fromStep);
+    }
+    this.hasCleanedFlag = true;
+  };
+
+  private generateRuntimeEnvPath = (step: string) => {
+    const envFilePostfix = "_env.json";
+    return path.join(this.runtimeFolder, ".runtime", step + envFilePostfix);
```
Comment (ruowan) [inline lib/testScenario/runtimeEnvManager.ts:112]: The env path is not this folder. It should be in `generated/<RP>/<testScenario>/<runId>/runtime/`. All generated file should be together

File: lib/commands/run-test-scenario.ts [hunk 4]
```diff
@@ -88,13 +90,32 @@ export const builder: yargs.CommandBuilder = {
...
+  },
+  to: {
+    describe:
+      "the step to end in current run,it's used for debugging and make sure not use --cleanUp to delete resource group in the previous run.",
+    string: true,
+    demandOption: false,
+    implies: "runId",
+  },
+  runId: {
+    describe: "specify the last runId for debugging",
```
Comment (ruowan) [inline lib/commands/run-test-scenario.ts:115]: Why the last? We could support any runId if current local env storage exist.

File: lib/testScenario/runtimeEnvManager.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+import { PostmanCollectionRunnerClientOption } from "./postmanCollectionRunnerClient";
+
+interface RuntimeEnvContainer {
+  afterStep?: { [key: string]: VariableDefinition }; // to save env when the item is completed, not used now.
+  beforeStep?: { [key: string]: VariableDefinition }; // to save env when the item starts.
+}
+
+export class RuntimeEnvManager {
+  private hasCleanedFlag = false;
+  private runtimeEnvCollection: Map<string, RuntimeEnvContainer> = new Map<
```
Comment (ruowan) [inline lib/testScenario/runtimeEnvManager.ts:14]: what's the runtimeEnvCollection? What's the key here. Better to have comment

File: lib/testScenario/postmanCollectionRunnerClient.ts (hunk 2 above)
Comment (jianyexi) [reply lib/testScenario/postmanCollectionRunnerClient.ts:544]: clean should trigger when any of step is completed

File: lib/commands/run-test-scenario.ts (hunk 4 above)
Comment (jianyexi) [reply lib/commands/run-test-scenario.ts:115]: Good capture, I'll remove the 'last' from the description

File: lib/testScenario/runtimeEnvManager.ts (hunk 3 above)
Comment (jianyexi) [reply lib/testScenario/runtimeEnvManager.ts:112]: 'this.runtimeFolder' is  same with the current report path ,  following  `generated/<RP>/api-version/<testScenarioFileName>/<runId>/<testScenarioName>` ,

**Comment by ruowan** [comment]
The CI build failed. Pls fix it.

**Review by ruowan — APPROVED** [review]

**Pull request merged**